components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/OAuthClientAuthnService.java

```diff
@@ -144,7 +157,63 @@ private void executeClientAuthenticators(HttpServletRequest request, OAuthClient
         if (log.isDebugEnabled()) {
             log.debug("Executing OAuth client authenticators.");
         }
-
+        try {
+            String clientId = null;
+            for (OAuthClientAuthenticator oAuthClientAuthenticator : this.getClientAuthenticators()) {
+                try {
+                    clientId = extractClientId(oAuthClientAuthenticator, request,
+                            bodyContentMap);
+                    if (StringUtils.isNotBlank(clientId)) {
+                        break;
+                    }
+                } catch (OAuthClientAuthnException e) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Client ID cannot be extracted using the " + oAuthClientAuthenticator.getName());
+                    }
+                }
+            }
+            if (StringUtils.isNotBlank(clientId)) {
```

Comment:
Lets remove the nested if condition and set the error on isBlank(clientId)

Reply:
Fixed in 51fdd762037ef49e2ffb96749e616e1a0343f202